apps/site/lib/site_web/controllers/schedule/line/helpers.ex

```diff
@@ -80,10 +80,36 @@ defmodule SiteWeb.ScheduleController.Line.Helpers do
   defp do_get_branch_route_stops(route, direction_id, route_pattern_id) do
     route.id
     |> get_route_patterns(direction_id, route_pattern_id)
-    |> Enum.filter(&by_typicality(&1, route_pattern_id))
+    |> filter_by_min_typicality()
+    # The below row is for testing purposes for the ferry ONLY
+    # This is needed until a specific routepattern has a distinct typicality
+    |> temporary_hh_ferry_only_func(route.id)
     |> Enum.map(&stops_for_route_pattern/1)
   end
 
+  @spec filter_by_min_typicality([RoutePattern.t()]) :: [RoutePattern.t()]
+  defp filter_by_min_typicality(route_patterns) do
+    route_patterns
+    |> Enum.reduce({nil, []}, fn
+      %RoutePattern{typicality: typicality} = r, {t, _a} when typicality < t -> {typicality, [r]}
+      %RoutePattern{typicality: typicality} = r, {typicality, _a} -> {typicality, [r | _a]}
```

Comment:
I'll admit I'm still trying to figure out what it's doing, I think something like grouping by typicality while also sorting by the lowest? It might be worth pulling this out into a separate function for clarity, or maybe using some more Enum functions like Enum.group_by, sort_by

Reply:
Ohhh hmmm.... Yes, I should pull it out into a function.  I'm curious about using other Enum grouping / sorting options - let's see what I can stitch together

Reply:
Perhaps I could use grouping, but it's not exactly what I'm doing here:  these lines populate an array to keep track of the lowest-typicality routePatterns.  I've expanded it into a separate function, but not sure I made it any clearer?